rust/ql/lib/codeql/rust/security/regex/RegexInjectionQuery.qll

```diff
@@ -0,0 +1,31 @@
+/**
+ * Provides a taint-tracking configuration for detecting regular expression
+ * injection vulnerabilities.
+ */
```

Comment:
We're generally going for a simpler two-files-per-query setup in Rust:
```
lib/.../FooExtensions.qll
src/.../Foo.ql
```
rather than three:
```
lib/.../FooExtensions.qll
lib/.../FooQuery.qll [this file]
src/.../Foo.ql
```
Before we had query inline expectations tests, the latter pattern was useful as the taint config in `FooQuery.qll` could be used to define a (non-query) inline expectations test.  These days there's less benefit to not just defining the taint configuration in the `.ql`, which I like because there are fewer files to navigate between.

So I'd prefer we combined this file into the `.ql`, though I don't feel particularly strongly if you'd rather we worry about things more consistent with other queries later on.

Reply:
That makes sense and fewer files seems great! I've updated the PR such that `RegexInjectionQuery.qll` is removed and now directly inside the query file.

Reply:
Great, thanks for updating the query!